src/contributing/prs.md

```diff
@@ -0,0 +1,150 @@
+     “Deal with an empty list correctly” is) and a description (empty
+     descriptions are almost never acceptable). If it fixes a GitHub issue, add
+     the text `Fixes #<GitHub issue number>` into the description.
+
+Your pull request has now been raised and will be reviewed by a yk contributor.
+The aim of a review is two fold: to ensure that contributions to yk are of an
+acceptable quality; and that at least two people (the reviewer and reviewee)
+understand what was changed and why.
+
+The reviewer will comment in detail on specific lines of the commit and on the
+overall pull request. Comments might be queries (e.g. “can the list ever be
+empty at this point?”) or corrections (e.g. “we need to deal with the
+possibility of an empty list at this point”). For each comment you can either:
+
+  1. Address the issue. For example, if the reviewer has pointed out correctly
+     that something needs to be changed, make a small additional commit, push
+     it to your branch and make a comment `Fixed in #<git hash>` so that the
+     reviewer can check that the fix addresses their point. Each fix should be
+     in a separate commit so that it is easy to associate a comments with its
+     fix. Note that you must *never* force push updates to your branch unless
+     explicitly requested to do so by the reviewer: always make new commits on
+     your existing branch.
+
+  2. Add a query. You might not understand the reviewer’s question: it’s fine
+     to ask for clarification.
+
+  3. Explain why you think there is not an issue. The reviewer might have
+     misunderstood either the details of a change, or the pull request’s
+     context (e.g. the pull request might not have intended to fix every
+     possible issue). A complete, and polite explanation, can help both
+     reviewer and reviewee focus on the essentials.
+
+Often multiple rounds of comments, queries, and fixes are required. When the
+reviewer is happy with the changes, they will ask the reviewee to “please
+squash” their pull request. The aim here is to provide a readable sequence of
+commits for those who later need to look at the history of changes to a
+particular part the repository. The general aim is to provide a pleasing
+sequence of self-contained commits which clearly articulate a change to future
+readers. At the very least, all of the “fix commits” must be merged away:
+commonly, these are merged into the main commits, or, though rarely, into a new
+commit(s). It is not required, and often is undesireable, to squash all commits
+down into a single commit: when multiple commits can better explain a change,
+they are much preferred.
+
+The process of squashing is synonymous with `git rebase`, so when you are asked
+to squash it is also acceptable to: rebase the commit against the `master`
+branch at the same time; and force push the resulting rebased branch. *This is
+the only time you should force push an update to a branch.*
```

Comment:
Can we add a step here: check that the commit descriptions still make sense and amend as necessary.

Reply:
Fixed in 1b9bbcb.